Dependency conflict with ClickHouse client in plugin-jdbc 0.24 on Kestra 0.24 (`NoSuchMethodError` in HttpComponents)

### Describe the issue

**Description:**
When using **Kestra 0.24** with **plugin-jdbc 0.24** and the ClickHouse JDBC driver, any query task fails due to a dependency conflict between the ClickHouse client and Apache HttpComponents libraries bundled with the plugin.

**Error log excerpt:**

```
2025-08-08T19:56:40.138035Z ERROR Could not initialize class com.clickhouse.client.api.internal.HttpAPIClientHelper$MeteredManagedHttpClientConnectionFactory
2025-08-08T19:56:40.138035Z ERROR Exception java.lang.NoSuchMethodError: 'void org.apache.hc.core5.http.impl.io.DefaultHttpRequestWriterFactory.<init>(org.apache.hc.core5.http.config.Http1Config)' [in thread "worker_7"]
2025-08-08T19:56:40.138035Z TRACE java.lang.NoClassDefFoundError: Could not initialize class com.clickhouse.client.api.internal.HttpAPIClientHelper$MeteredManagedHttpClientConnectionFactory
	at com.clickhouse.client.api.internal.HttpAPIClientHelper.poolConnectionManager(HttpAPIClientHelper.java:226)
	at com.clickhouse.client.api.internal.HttpAPIClientHelper.createHttpClient(HttpAPIClientHelper.java:304)
	at com.clickhouse.client.api.internal.HttpAPIClientHelper.<init>(HttpAPIClientHelper.java:114)
	at com.clickhouse.client.api.Client.<init>(Client.java:176)
	at com.clickhouse.client.api.Client.<init>(Client.java:115)
	at com.clickhouse.client.api.Client$Builder.build(Client.java:1064)
	at com.clickhouse.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:99)
	at com.clickhouse.jdbc.Driver.connect(Driver.java:123)
	at com.clickhouse.jdbc.ClickHouseDriver.connect(ClickHouseDriver.java:97)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at io.kestra.plugin.jdbc.JdbcConnectionInterface.connection(JdbcConnectionInterface.java:102)
	at io.kestra.plugin.jdbc.AbstractJdbcQuery.run(AbstractJdbcQuery.java:41)
	at io.kestra.plugin.jdbc.clickhouse.Query.run(Query.java:20)
	at io.kestra.core.runners.WorkerTaskCallable.doCall(WorkerTaskCallable.java:79)
	at io.kestra.core.runners.AbstractWorkerCallable.call(AbstractWorkerCallable.java:63)
	at io.kestra.core.runners.WorkerSecurityService.callInSecurityContext(WorkerSecurityService.java:10)
	at io.kestra.core.runners.Worker.lambda$callJob$20(Worker.java:968)
	at io.kestra.core.trace.NoopTracer.inCurrentContext(NoopTracer.java:15)
	at io.kestra.core.runners.Worker.callJob(Worker.java:964)
	at io.kestra.core.runners.Worker.runAttempt(Worker.java:922)
	at io.kestra.core.runners.Worker.run(Worker.java:733)
	at io.kestra.core.runners.Worker.handleTask(Worker.java:381)
	at io.kestra.core.runners.Worker.lambda$run$8(Worker.java:296)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoSuchMethodError: 'void org.apache.hc.core5.http.impl.io.DefaultHttpRequestWriterFactory.<init>(org.apache.hc.core5.http.config.Http1Config)' [in thread "worker_7"]
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<init>(ManagedHttpClientConnectionFactory.java:86)
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<init>(ManagedHttpClientConnectionFactory.java:105)
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<init>(ManagedHttpClientConnectionFactory.java:120)
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<init>(ManagedHttpClientConnectionFactory.java:127)
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<init>(ManagedHttpClientConnectionFactory.java:131)
	at org.apache.hc.client5.http.impl.io.ManagedHttpClientConnectionFactory.<clinit>(ManagedHttpClientConnectionFactory.java:66)
	... 28 more

```

**Steps to reproduce:**

1. Install Kestra 0.24.
2. Configure a ClickHouse JDBC connection (using `clickhouse-jdbc` with `clickhouse-client` API).
3. Run any SQL query task.
4. Observe the error.

**Expected behavior:**
The ClickHouse JDBC connection should initialize and execute queries without dependency conflicts.

**Actual behavior:**
The task fails immediately due to a `NoSuchMethodError` caused by mismatched versions of `org.apache.hc.core5.http` classes between the ClickHouse client and the dependencies packaged in plugin-jdbc.

**Additional context:**
It seems the ClickHouse client requires a newer version of Apache HttpComponents (`httpcore5` / `httpclient5`) than the one shaded or provided by the plugin. This might require aligning the dependency versions or isolating the ClickHouse client dependencies to prevent conflicts.

### Environment

- Kestra Version: 0.24
- Kestra Plugin JDBC Version: 0.24
- Deployment: Kubernetes standalone

--- a/plugin-jdbc-clickhouse/build.gradle
+++ b/plugin-jdbc-clickhouse/build.gradle
@@ -13,7 +13,14 @@ jar {
 }
 
 dependencies {
-    implementation 'com.clickhouse:clickhouse-jdbc:0.8.6'
+    implementation('com.clickhouse:clickhouse-jdbc:0.8.6') {
+        exclude group: 'org.apache.httpcomponents.core5'
+        exclude group: 'org.apache.httpcomponents.client5'
+    }
+
+    implementation 'org.apache.httpcomponents.core5:httpcore5:5.2.4'
+    implementation 'org.apache.httpcomponents.core5:httpcore5-h2:5.2.4'
+    implementation 'org.apache.httpcomponents.client5:httpclient5:5.3.1'
     implementation 'org.lz4:lz4-java:1.8.0'
 
     implementation project(':plugin-jdbc')

